### PR TITLE
feat: Initialize Fine-Tuning Interface and Core Methods - Part 3

### PR DESCRIPTION
**Pull request by ishaansehgal99**
Setup

### CODE REVIEW

File: pkg/model/interface.go [hunk 1]
```diff
@@ -7,12 +7,13 @@ import (
...
-	GetInferenceParameters() *PresetInferenceParam
+	GetInferenceParameters() *PresetParam
+	GetTrainingParameters() *PresetParam
 	SupportDistributedInference() bool //If true, the model workload will be a StatefulSet, using the torch elastic runtime framework.
 }
 
-// PresetInferenceParam defines the preset inference parameters for a model.
-type PresetInferenceParam struct {
+// PresetParam defines the preset inference parameters for a model.
+type PresetParam struct {
```
Comment (Fei-Guo) [inline pkg/model/interface.go:16]: 1) Move BaseCommand to be adjacent to ModelRunParams;
2) Rename DeploymentTimeout-> ImagepullTimeout;
3) refresh all comments to explicitly;
4) Add a supporttuning interface method, used by validation.
// Llama does not support.

File: api/v1alpha1/workspace_condition_types.go [hunk 2]
```diff
@@ -16,6 +16,9 @@ const (
 	// WorkspaceConditionTypeInferenceStatus is the state when Inference has been created.
 	WorkspaceConditionTypeInferenceStatus = ConditionType("InferenceReady")
 
+	// WorkspaceConditionTypeTuningStatus is the state when Tuning has been created.
+	WorkspaceConditionTypeTuningStatus = ConditionType("TuningReady")
```
Comment (Fei-Guo) [inline api/v1alpha1/workspace_condition_types.go:20]: For job type workload, the status cannot be "READY".  The status can be "STARTED" and "DONE". Check what status code the build-in k8s job uses.

File: pkg/controllers/workspace_controller.go [hunk 3]
```diff
@@ -109,16 +110,27 @@ func (c *WorkspaceReconciler) addOrUpdateWorkspace(ctx context.Context, wObj *ka
...
-			return reconcile.Result{}, updateErr
+	if wObj.Tuning != nil {
+		if err = c.applyTuning(ctx, wObj); err != nil {
+			if updateErr := c.updateStatusConditionIfNotMatch(ctx, wObj, kaitov1alpha1.WorkspaceConditionTypeReady, metav1.ConditionFalse,
+				"workspaceFailed", err.Error()); updateErr != nil {
+				klog.ErrorS(updateErr, "failed to update workspace status", "workspace", klog.KObj(wObj))
+				return reconcile.Result{}, updateErr
+			}
+			return reconcile.Result{}, err
+		}
```
Comment (Fei-Guo) [inline pkg/controllers/workspace_controller.go:121]: We need to carefully think about how to update the workspace state after tunning job is done. (it cannot be READY). But we can deal with it follow up PRs.

File: pkg/controllers/workspace_controller.go [hunk 4]
```diff
@@ -423,6 +435,54 @@ func (c *WorkspaceReconciler) ensureService(ctx context.Context, wObj *kaitov1al
...
+func (c *WorkspaceReconciler) applyTuning(ctx context.Context, wObj *kaitov1alpha1.Workspace) error {
+	var err error
+	func() {
+		if wObj.Tuning.Preset != nil {
+			presetName := string(wObj.Tuning.Preset.Name)
+			model := plugin.KaitoModelRegister.MustGet(presetName)
+
+			trainingParam := model.GetTrainingParameters()
+
+			existingObj := &appsv1.Deployment{}
```
Comment (Fei-Guo) [inline pkg/controllers/workspace_controller.go:447]: This cannot be deployment. It needs to be Job.

**Review by Fei-Guo — COMMENTED** [review]
Replace all "training" with "tuning".  

We can deal with workspace conditions in future changes.

File: pkg/controllers/workspace_controller.go (hunk 4 above)
Comment (ishaansehgal99) [reply pkg/controllers/workspace_controller.go]: Done

File: api/v1alpha1/workspace_condition_types.go (hunk 2 above)
Comment (ishaansehgal99) [reply api/v1alpha1/workspace_condition_types.go]: Done

File: pkg/model/interface.go [hunk 5]
```diff
@@ -7,27 +7,29 @@ import (
...
+	// BaseCommand is the initial command (e.g., 'torchrun', 'accelerate launch') used in the command line.
+	BaseCommand    string
+	ModelRunParams map[string]string // Parameters for running the model training/inference.
+	// WorkloadTimeout defines the maximum duration for creating the workload.
 	// This timeout accommodates the size of the image, ensuring pull completion
 	// even under slower network conditions or unforeseen delays.
-	DeploymentTimeout time.Duration
-	// BaseCommand is the initial command (e.g., 'torchrun', 'accelerate launch') used in the command line.
-	BaseCommand string
+	WorkloadTimeout time.Duration
```
Comment (Fei-Guo) [inline pkg/model/interface.go:32]: Rename workloadTimeout -> ReadinessTimeout?

File: pkg/tuning/preset-tuning.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+	kaitov1alpha1 "github.com/azure/kaito/api/v1alpha1"
+	"github.com/azure/kaito/pkg/model"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreatePresetTuning(ctx context.Context, workspaceObj *kaitov1alpha1.Workspace,
+	tuningObj *model.PresetParam, kubeClient client.Client) (client.Object, error) {
+	// TODO
+
+	// e.g. example from Inference
```
Comment (Fei-Guo) [inline pkg/tuning/preset-tuning.go:14]: remove commented code.

File: api/v1alpha1/workspace_condition_types.go [hunk 7]
```diff
@@ -16,6 +16,15 @@ const (
 	// WorkspaceConditionTypeInferenceStatus is the state when Inference has been created.
 	WorkspaceConditionTypeInferenceStatus = ConditionType("InferenceReady")
 
+	// WorkspaceConditionTypeTuningStarted indicates that the tuning Job has been started.
+	WorkspaceConditionTypeTuningStarted = ConditionType("TuningStarted")
+
+	// WorkspaceConditionTypeTuningComplete indicates that the tuning Job has completed successfully.
```
[the remark this answers is not in the document]
Comment (ishaansehgal99) [reply api/v1alpha1/workspace_condition_types.go]: Cool, I removed these for future

File: pkg/controllers/workspace_controller.go [hunk 8]
```diff
@@ -109,16 +111,27 @@ func (c *WorkspaceReconciler) addOrUpdateWorkspace(ctx context.Context, wObj *ka
...
 	}
 
-	if err = c.applyInference(ctx, wObj); err != nil {
-		if updateErr := c.updateStatusConditionIfNotMatch(ctx, wObj, kaitov1alpha1.WorkspaceConditionTypeReady, metav1.ConditionFalse,
-			"workspaceFailed", err.Error()); updateErr != nil {
-			klog.ErrorS(updateErr, "failed to update workspace status", "workspace", klog.KObj(wObj))
-			return reconcile.Result{}, updateErr
+	if wObj.Tuning != nil {
+		if err = c.applyTuning(ctx, wObj); err != nil {
+			if updateErr := c.updateStatusConditionIfNotMatch(ctx, wObj, kaitov1alpha1.WorkspaceConditionTypeReady, metav1.ConditionFalse,
```
[the remark this answers is not in the document]
Comment (ishaansehgal99) [reply pkg/controllers/workspace_controller.go]: Done

File: pkg/model/interface.go (hunk 5 above)
Comment (ishaansehgal99) [reply pkg/model/interface.go]: Done

File: pkg/tuning/preset-tuning.go (hunk 6 above)
Comment (ishaansehgal99) [reply pkg/tuning/preset-tuning.go]: Done

**Review by Fei-Guo — APPROVED** [review]

**Pull request merged**